### PR TITLE
revirew my code

### PR DESCRIPTION
**Pull request by AlexanderRusak**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,231 +1,138 @@
...
+
+  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
+    integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous" />
+  <script src="https://kit.fontawesome.com/aa7a31f950.js" crossorigin="anonymous"></script>
+  <style>
+    .wrapper {
+      min-height: calc(100vh - 15px);
+      margin: 15px auto 0;
+    }
+  </style>
```
Comment (maximsan) [inline index.html]: move into style.css

File: index.html [hunk 2]
```diff
@@ -1,231 +1,138 @@
...
+  <meta charset="utf-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
+
+  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
+    integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous" />
+  <script src="https://kit.fontawesome.com/aa7a31f950.js" crossorigin="anonymous"></script>
+  <style>
+    .wrapper {
+      min-height: calc(100vh - 15px);
+      margin: 15px auto 0;
```
Comment (maximsan) [inline index.html]: change to rem

File: index.html [hunk 3]
```diff
@@ -1,231 +1,138 @@
...
+    }
+  </style>
+  <title>Death Valley 2</title>
+</head>
+
+<body id="body" cz-shortcut-listen="true">
```
Comment (maximsan) [inline index.html]: remove id="body" cz-shortcut-listen="true"

File: src/scripts/script.js [hunk 4]
```diff
@@ -1,34 +1,38 @@
...
+            <h5 class="mb-1">{{Title}}</h5>
         </div>
-        <p class="mb-1 w-100">${text}</p>
+        <p class="mb-1 w-100">{{Text}}</p>
     </div>
-        ${!status ? getCurrentAdditionalTemplate() : ""}`);
-}
+        ${!status ? getCurrentAdditionalTemplate() : ""}`
+  );
+  return taskTemplate({ Title: title, Text: text });
```
Comment (maximsan) [inline src/scripts/script.js]: return taskTemplate({ title, text });

File: src/scripts/script.js [hunk 5]
```diff
@@ -38,179 +42,201 @@ const getCurrentAdditionalTemplate = () => {
...
+    </div>`;
+};
 const removeToDoItem = (selectedNode) => {
-    const key = selectedNode.parentNode.parentNode.parentNode.id;
-    const deletingElement = getElementById(key);
-    deletingElement.parentNode.removeChild(deletingElement);
-    localStorage.removeItem(+key);
-    getCountTodos();
-}
+  const key = selectedNode.parentNode.parentNode.parentNode.id;
```
Comment (maximsan) [inline src/scripts/script.js]: rewrite this

File: src/scripts/script.js [hunk 6]
```diff
@@ -38,179 +42,201 @@ const getCurrentAdditionalTemplate = () => {
...
-            }
-            if (current.id < next.id) {
-                return !compareCore ? 1 : -1;
-            }
-            if (current.id == next.id) {
-                return 0;
-            }
-        }).forEach(node => currentContainer.appendChild(node));
+  const currentContainer = getElementById("currentTasks");
+  const compareCore = compare == "sort-from-new" ? true : false;
```
Comment (maximsan) [inline src/scripts/script.js]: `const compareCore = compare === "sort-from-new"`

File: src/scripts/script.js [hunk 7]
```diff
@@ -38,179 +42,201 @@ const getCurrentAdditionalTemplate = () => {
...
+      if (current.id > next.id) {
+        return compareCore ? 1 : -1;
+      }
+      if (current.id < next.id) {
+        return !compareCore ? 1 : -1;
+      }
+      if (current.id == next.id) {
+        return 0;
+      }
+    })
```
Comment (maximsan) [inline src/scripts/script.js]: ectract method

File: src/scripts/script.js [hunk 8]
```diff
@@ -38,179 +42,201 @@ const getCurrentAdditionalTemplate = () => {
...
 
+const setCountTodos = () => {
+  let countTodos = [];
+  let completedTasks = 0;
+  let currentTask = 0;
+  for (let i = 0; i < localStorage.length; i++) {
+    countTodos.push(localStorage.key(i));
+    JSON.parse(localStorage.getItem(countTodos[i])).isCompleted == true
+      ? completedTasks++
+      : currentTask++;
```
Comment (maximsan) [inline src/scripts/script.js]: rewrite it

File: src/scripts/script.js [hunk 9]
```diff
@@ -38,179 +42,201 @@ const getCurrentAdditionalTemplate = () => {
...
+  }
+  getElementById(
+    "todosCount"
+  ).innerText = ` ${currentTask} / ${completedTasks}`;
```
Comment (maximsan) [inline src/scripts/script.js]: 1) get component 2) apply styles and whatever else you want

File: src/scripts/script.js [hunk 10]
```diff
@@ -38,179 +42,201 @@ const getCurrentAdditionalTemplate = () => {
...
-        for (let i = 0; i < fontElement.length; i++) {
-            fontElement[i].style.color = "#fff";
-        }
-
-    } else {
-        getElementById("body").style.backgroundColor = "#fff";
-        for (let i = 0; i < fontElement.length; i++) {
-            fontElement[i].style.color = "#000";
-        }
+getElementById("customSwitch1").addEventListener("change", function () {
```
Comment (maximsan) [inline src/scripts/script.js]: rename `customSwitch1`

File: src/scripts/script.js [hunk 11]
```diff
@@ -38,179 +42,201 @@ const getCurrentAdditionalTemplate = () => {
...
+  const fontElement = document.querySelectorAll("h3");
+  if (this.checked) {
+    getElementById("body").style.backgroundColor = "#000";
+    for (let i = 0; i < fontElement.length; i++) {
+      fontElement[i].style.color = "#fff";
     }
```
Comment (maximsan) [inline src/scripts/script.js]: use forEach

File: src/scripts/script.js [hunk 12]
```diff
@@ -38,179 +42,201 @@ const getCurrentAdditionalTemplate = () => {
...
-
+  } else {
+    getElementById("body").style.backgroundColor = "#fff";
+    for (let i = 0; i < fontElement.length; i++) {
+      fontElement[i].style.color = "#000";
+    }
```
Comment (maximsan) [inline src/scripts/script.js]: use forEach

File: src/scripts/script.js [hunk 13]
```diff
@@ -38,179 +42,201 @@ const getCurrentAdditionalTemplate = () => {
...
-
-    } else {
-        getElementById("body").style.backgroundColor = "#fff";
-        for (let i = 0; i < fontElement.length; i++) {
-            fontElement[i].style.color = "#000";
-        }
+getElementById("customSwitch1").addEventListener("change", function () {
+  const fontElement = document.querySelectorAll("h3");
+  if (this.checked) {
+    getElementById("body").style.backgroundColor = "#000";
```
Comment (maximsan) [inline src/scripts/script.js]: move in global scope

**Review by maximsan — CHANGES_REQUESTED** [review]
I left notes. Fix please them.

File: index.html (hunk 2 above)
Comment (AlexanderRusak) [reply index.html]: changed

File: dist/bundle.js [hunk 14]
```diff
@@ -7,15 +7,325 @@
  * If you are looking for production-ready output files, see mode: "production" (https://webpack.js.org/configuration/mode/).
```
Comment (maximsan) [inline dist/bundle.js]: @AlexanderRusak don't commit dist folder. It doesn't make sense. You should used to keep it ignored in .gitignore file. Please, delete from PR

File: dist/style.css [hunk 15]
```diff
@@ -1 +1,4 @@
+.wrapper {
+  min-height: calc(100vh - 15px);
+  margin: 1rem auto 0; }
```
Comment (maximsan) [inline dist/style.css]: the same with this file. And why do you have it as not minified file in dist folder ? You should change your webpack setup. We use `dist` folder as a production version of our project. In that case we need all minified, uglified and I believe we don't need source-maps for `css/scss`.

File: src/scripts/script.js [hunk 16]
```diff
@@ -38,179 +42,200 @@ const getCurrentAdditionalTemplate = () => {
...
-            }
-            if (current.id < next.id) {
-                return !compareCore ? 1 : -1;
-            }
-            if (current.id == next.id) {
-                return 0;
-            }
-        }).forEach(node => currentContainer.appendChild(node));
+  const currentContainer = getElementById("currentTasks");
+  const compareCore = compare === "sort-from-new" ? true : false;
```
Comment (maximsan) [inline src/scripts/script.js]: it's the same `const compareCore = compare === "sort-from-new"`

File: src/scripts/script.js [hunk 17]
```diff
@@ -38,179 +42,200 @@ const getCurrentAdditionalTemplate = () => {
...
+      if (current.id > next.id) {
+        return compareCore ? 1 : -1;
+      }
+      if (current.id < next.id) {
+        return !compareCore ? 1 : -1;
+      }
+      if (current.id == next.id) {
+        return 0;
+      }
+    })
```
Comment (maximsan) [inline src/scripts/script.js]: you can write it like `.sort(sortById(current.Id, next.Id)`
and then
`sortById = (currentId, nextId) => { 
    return currentId === nextId ? 0 : currentId > nextId ? -1 : 1 
}`

File: src/scripts/script.js [hunk 18]
```diff
@@ -38,179 +42,200 @@ const getCurrentAdditionalTemplate = () => {
...
+function toggleStyle() {
+  const fontElement = document.querySelectorAll("h3");
+  if (this.checked) {
+    document.getElementsByTagName("body")[0].style.backgroundColor = "#000";
+    fontElement.forEach((felement) => {
+      felement.style.color = "#fff";
+    });
+  } else {
+    document.getElementsByTagName("body")[0].style.backgroundColor = "#fff";
+    fontElement.forEach((felement) => {
```
Comment (maximsan) [inline src/scripts/script.js]: follow `camelCase` style in naming -> `fElement or fontElement or fontElem`

File: src/scripts/script.js [hunk 19]
```diff
@@ -38,179 +42,200 @@ const getCurrentAdditionalTemplate = () => {
...
+      id_dateValue,
+      ...getModalValues(),
+      false
+    );
+  });
+};
+function toggleStyle() {
+  const fontElement = document.querySelectorAll("h3");
+  if (this.checked) {
+    document.getElementsByTagName("body")[0].style.backgroundColor = "#000";
```
Comment (maximsan) [inline src/scripts/script.js]: create variable for constants. what you do here is a `magic code`. 
1) https://stackoverflow.com/questions/41725725/access-css-variable-from-javascript - you can use this approach
2) u can create variables like `--main-bg-color = "#000"` and then call it in code 
3) also you can change body styles like this `document.body.style.background = --main-bg-color`
https://stackoverflow.com/questions/197748/how-do-i-change-the-background-color-with-javascript

File: src/scripts/script.js [hunk 20]
```diff
@@ -38,179 +42,200 @@ const getCurrentAdditionalTemplate = () => {
...
+  ).value;
+  return [input_titleValue, input_textValue, input_radioValue];
+};
 const setModalValues = (title, text, priority) => {
-    getElementById("inputTitle").value = title;
-    getElementById("inputText").value = text;
-    getElementById(`${priority}`).setAttribute("checked", "checked");
-}
+  getElementById("inputTitle").value = title;
+  getElementById("inputText").value = text;
```
Comment (maximsan) [inline src/scripts/script.js]: you have this variable in several places. think how to initialize them once and reuse in `getModalValues `, `setModalValues ` and etc.

**Review by maximsan — CHANGES_REQUESTED** [review]
Hello. I left some new comments. Please adjust webpack setup + use ESM modules for js code. Also do not skip my comments, all of them should be answered. Thanks!

File: dist/bundle.js (hunk 14 above)
Comment (AlexanderRusak) [reply dist/bundle.js]: it has been done

File: dist/style.css (hunk 15 above)
Comment (AlexanderRusak) [reply dist/style.css]: i have removed dist folder from pr but if someone is want to minified  files he will use npm run build to build project and minified files

File: src/scripts/script.js (hunk 16 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: fixed

File: src/scripts/script.js (hunk 17 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: i have had absolutely different logic in that place.
I must use third parameter for detected events on buttons.
In that place i toggled beetween up-sort or down-sort in the same function

File: src/scripts/script.js (hunk 18 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: fixed)

File: src/scripts/script.js (hunk 19 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: Good option for future, i've fixed that

File: src/scripts/script.js (hunk 20 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: getModalValues  setModalValues this are functions and they are using in several places!
Maybe i didn't correct understand you. Did you mean getElementById("inputTitle").value = title;  getElementById("inputText").value = text???

File: src/scripts/script.js (hunk 13 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: has been moved

File: src/scripts/script.js (hunk 12 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: has been done

File: src/scripts/script.js (hunk 11 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: has been done

File: src/scripts/script.js (hunk 10 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: has been renamed

File: src/scripts/script.js (hunk 9 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: has corrected

File: src/scripts/script.js (hunk 5 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: Has been rewritten

File: src/scripts/script.js (hunk 4 above)
Comment (AlexanderRusak) [reply src/scripts/script.js]: fixed that

File: index.html (hunk 3 above)
Comment (AlexanderRusak) [reply index.html]: has been removed that

File: index.html (hunk 1 above)
Comment (AlexanderRusak) [reply index.html]: has been moved

File: dist/bundle.js (hunk 14 above)
Comment (AlexanderRusak) [reply dist/bundle.js]: I have been rewritten webpack configuration file.
Add to .gitignore dist folder and then remove from pr

File: dist/style.css (hunk 15 above)
Comment (AlexanderRusak) [reply dist/style.css]: i've added to config file build script which provides minified files in production version just use npm run build and will see magic like Amayak Akopyan

File: src/scripts/script.js (hunk 18 above)
Comment (AlexanderRusak) [reply src/scripts/script.js:181]: renamed

File: src/scripts/script.js (hunk 20 above)
Comment (maximsan) [reply src/scripts/script.js]: `const input_titleValue = getElementById("inputTitle").value`
`getElementById("inputTitle").value = title` and etc...

**Comment by AlexanderRusak** [comment]
I've resolved

**Review by maximsan — APPROVED** [review]
Fine

**Review by Larkgronick — APPROVED** [review]

**Pull request merged**